api/server/router/network/network_routes.go

```diff
@@ -70,14 +72,62 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 		return err
 	}
 
-	nw, err := n.backend.FindNetwork(vars["id"])
-	if err != nil {
-		if nr, err := n.cluster.GetNetwork(vars["id"]); err == nil {
-			return httputils.WriteJSON(w, http.StatusOK, nr)
+	// In case multiple networks have duplicate names, return error.
+	// TODO (yongtang): should we wrap with version here for backward compatibility?
+
+	// Find based on full ID
+	nr, _ := n.cluster.GetNetworks()
+	for _, network := range nr {
+		if network.ID == vars["id"] {
+			return httputils.WriteJSON(w, http.StatusOK, network)
+		}
+	}
+	nw := n.backend.GetNetworks()
+	for _, network := range nw {
+		if network.ID() == vars["id"] {
+			return httputils.WriteJSON(w, http.StatusOK, n.buildNetworkResource(network))
+		}
+	}
```

Comment:
Looks like `n.cluster` now takes precedence over `n.backend`. It was the other way around in the old code. Is it okay to change the order?